core/src/main/scala/chisel3/properties/Class.scala

```diff
@@ -206,6 +206,22 @@ object Class {
       val classType = ClassType.unsafeGetClassTypeByName(baseModule.name)
       Property[classType.Type]()
     }
+
+    /** Get a ClassType type from a Definition[Class].
+      *
+      * This is useful when a ClassType type is needed for other Property types.
+      *
+      * This method is safe, and should be used over unsafeGetClassTypeByName when possible.
+      */
+    def getClassType: ClassType = {
+      // Get the BaseModule for the Class this is a definition of.
+      val baseModule = definition.getInnerDataContext.getOrElse(
+        throwException("Internal Error! Class instance did not have an associated BaseModule.")
```

Comment:
Obviously we don't expect to hit internal errors, but this is Chisel so we probably will eventually 😂. Please add some additional information to the error message, at least `definition`.

Reply:
We use this same error in a few places here so i'll update all three while i'm at it.